mypyc/build.py

```diff
@@ -40,23 +40,31 @@
 from mypyc.namegen import exported_name
 from mypyc.options import CompilerOptions
 
-if TYPE_CHECKING:
-    from distutils.core import Extension as _distutils_Extension
-    from typing_extensions import TypeAlias
+if sys.version_info < (3, 12):
+    if TYPE_CHECKING:
+        from distutils.core import Extension as _distutils_Extension
+        from typing_extensions import TypeAlias
 
-    from setuptools import Extension as _setuptools_Extension
+        from setuptools import Extension as _setuptools_Extension
 
-    Extension: TypeAlias = Union[_setuptools_Extension, _distutils_Extension]
+        Extension: TypeAlias = Union[_setuptools_Extension, _distutils_Extension]
 
-
-try:
-    # Import setuptools so that it monkey-patch overrides distutils
+    try:
+        # Import setuptools so that it monkey-patch overrides distutils
+        import setuptools
+    except ImportError:
+        pass
+    from distutils import ccompiler, sysconfig
+else:
     import setuptools
-except ImportError:
-    if sys.version_info >= (3, 12):
-        # Raise on Python 3.12, since distutils will go away forever
-        raise
-from distutils import ccompiler, sysconfig
+    from setuptools import Extension
+    from setuptools._distutils import (
+        ccompiler as _ccompiler,  # type: ignore[attr-defined]
+        sysconfig as _sysconfig,  # type: ignore[attr-defined]
+    )
+
+    ccompiler = _ccompiler
+    sysconfig = _sysconfig
```

Comment:
Use assignment as a workaround to avoid errors about unimported `Any`s.

Reply:
I opened https://github.com/python/typeshed/pull/10401 to help with this